rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/frontend/topology/TopologyImpl.java

```diff
@@ -94,7 +93,7 @@ public Long getDistance(InetAddress host, InetAddress host2) {
         } else if (distances.get(host2) != null && distances.get(host2).get(host) != null) {
```

Comment:
I know it is not related to your changes but as an enhancement, `distances.get(host)` and `distances.get(host2)` could be extracted into variables to avoid calling `get` twice. 

Besides, `new Long(0)` and  `new Long(Long.MAX_VALUE);` could maybe be cached to avoid creating multiple objects? if I am not wrong getDistance could be expensive since it is part of an algorithm whose time complexity is at least quadratic?